cylc/flow/taskdef.py

```diff
@@ -325,6 +325,21 @@ def get_parent_points(self, point):
                         parent_points.add(trig.get_parent_point(point))
         return parent_points
 
+    def get_prereqs(self, point):
+        """Return my prereqs, at point."""
+        prereqs = set()
+        for seq in self.sequences:
+            if not seq.is_valid(point):
+                continue
+            if seq in self.dependencies:
+                # task has prereqs in this sequence
+                for dep in self.dependencies[seq]:
+                    # TODO?
+                    if dep.suicide:
+                        continue
+                    prereqs.add(dep.get_prerequisite(point, self))
```

Comment:
Less nested, and probably faster

```suggestion
        for seq, (seq2, dependencies) in itertools.product(
            self.sequences, self.dependencies.items()
        ):
            if not seq == seq2 or not seq.is_valid(point):
                continue
            [
                prereqs.add(dep.get_prerequisite(point, self))
                for dep in dependencies
                if not dep.suicide
            ]
```